elastic/kibana
# [SLM] Hide system indices in snapshot restore flow

Partially addresses https://github.com/elastic/kibana/issues/120314

### Summary
The restore snapshot wizard lets you select system indices to restore. However, a restore request that includes a system index returns an error. Since we currently use the [get snapshot api](https://www.elastic.co/guide/en/elasticsearch/reference/current/get-snapshot-api.html), we can leverage the `feature_states` array (which gives us a list of system indices in the snapshot) in order to create a new array of "normal" indices that don't include any system ones.

**Steps to test:**
1. Run elasticsearch with `yarn es snapshot --license=trial -E path.repo=/tmp/es-backups` and also run kibana
2. Navigate to console and run the following command to create a repository:
```
.PUT /_snapshot/my_backup
{
  "type": "fs",
  "settings": {
    "location": "/tmp/es-backups",
    "chunk_size": "10mb"
  }
}
```
3. Go to the homepage and add sample data so that we have some indices to play with
4. Go to snapshot and restore app and create a new policy and leave all default values as they are. Run the policy.
7. Go to the snapshots tab and verify that in the details flyout there are no system indices listed
8. Verify that clicking restore and disabling the toggle from the `Data streams and indices` section also doesn't list any system indices.

Note: here's a list of system index patterns https://github.com/elastic/elasticsearch/issues/50251
